src/Compilers/Shared/CoreClrBuildClient.cs

```diff
@@ -46,14 +46,22 @@ protected override string GetSessionKey(BuildPaths buildPaths)
             return string.Empty;
         }
 
-        protected override async Task<BuildResponse> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string pipeName, string keepAlive, string libDirectory, CancellationToken cancellationToken)
+        protected override async Task<RunServerResult> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string pipeName, string keepAlive, string libDirectory, CancellationToken cancellationToken)
```

Comment:
It looks like this implementation actually can return a Task in an incomplete state.